### PR TITLE
SS-227 subscribe_to_plan should only respond with new subscription item.

### PR DESCRIPTION
**Pull request by spiropulo**

### CODE REVIEW

File: subhub/tests/unit/customer/test_customer_calls.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+logging.basicConfig(level=logging.DEBUG)
+
+
+class MockCustomer:
+    id = None
+    object = "customer"
+    subscriptions = [{"data": "somedata"}]
+
+
+def getFile(filename):
```
Comment (sidler-mozilla) [inline subhub/tests/unit/customer/test_customer_calls.py]: remember not camelCase but instead underscore_style.

File: subhub/tests/unit/customer/test_customer_calls.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    with open(os.path.join(os.path.realpath(os.path.dirname(__file__)), filename)) as f:
+        return json.load(f)
+
+
+def test_subscribe_to_plan_returns_newest(monkeypatch):
+    uid = uuid.uuid4()
+
+    subhubAccount = Mock(return_value=MockSubhubAccount())
```
Comment (sidler-mozilla) [inline subhub/tests/unit/customer/test_customer_calls.py]: change camelCase

File: subhub/tests/unit/customer/test_customer_calls.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+logging.basicConfig(level=logging.DEBUG)
+
+
+class MockCustomer:
+    id = None
+    object = "customer"
+    subscriptions = [{"data": "somedata"}]
+
+
+def getFile(filename):
```
Comment (sidler-mozilla) [inline subhub/tests/unit/customer/test_customer_calls.py]: getFile -> get_file

File: subhub/api/payments.py [hunk 4]
```diff
@@ -146,6 +139,40 @@ def sources_to_remove(subscriptions: list, customer: str) -> None:
...
 
 
+def cancel_subscription(uid, sub_id) -> FlaskResponse:
+    """
+    Cancel an existing subscription for a user.
+    :param uid:
+    :param sub_id:
+    :return: Success or failure message for the cancellation.
+    """
+    # TODO Remove payment source on cancel
```
[the remark this answers is not in the document]
Comment (marty331) [reply subhub/api/payments.py]: I believe this has been done already.

File: subhub/api/payments.py [hunk 5]
```diff
@@ -146,6 +137,39 @@ def sources_to_remove(subscriptions: list, customer: str) -> None:
...
+        return {"message": "Customer does not exist."}, 404
+    customer = stripe.Customer.retrieve(subscription_user.custId)
+    for item in customer["subscriptions"]["data"]:
+        if item["id"] == sub_id and item["status"] in [
+            "active",
+            "trialing",
+            "incomplete",
+        ]:
+            try:
+                stripe.Subscription.retrieve(sub_id)
```
Comment (sidler-mozilla) [inline subhub/api/payments.py]: looking at the diff; is this statement even required?

Comment (marty331) [reply subhub/api/payments.py]: I concur, please remove line 158.

**Pull request merged**